### PR TITLE
Improve sync result messages. Closes #1486

### LINKED ISSUE #1486: hard to view the complete error message on SyncFailed
**Describe the bug**
On Sync failure, the error message is truncated in the UI. It'd be nice if I could hover over it for the full message, or if it was multi-line text.

in the UI, i see `Warning: kubectl apply should be used on resource created by e...`

As I wrote this issue... I realize when I double-click select all of the text field  and cntrl-C, the entire error message gets added to the clipboard.

The full error message is : Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply The Deployment "pdf-service" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"release":"test-pdf-service", "app":"pdf-service"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable

**To Reproduce**
1. attempt to sync a manifest in a way that breaks things... In my case, I was changing selector labels in subsequent manifests

**Expected behavior**
It'd be nice to have a tooltip on hover that showed the full error.

**Screenshots**
Screen Shot 2019-04-17 at 5 50 40 PM

### PR DESCRIPTION
**Pull request by alexec**
See #1486 for screenshots.
Close #1486, #1624, #1651.

### CODE REVIEW

File: test/e2e/fixture/app/actions.go [hunk 1]
```diff
@@ -101,11 +119,7 @@ func (a *Actions) Refresh(refreshType RefreshType) *Actions {
 }
 
 func (a *Actions) Delete(cascade bool) *Actions {
-	args := []string{"app", "delete", a.context.name}
-	if cascade {
-		args = append(args, "--cascade")
-	}
-	a.runCli(args...)
+	a.runCli("app", "delete", a.context.name, fmt.Sprintf("--cascade=%v", cascade))
```
Comment (alexec) [inline test/e2e/fixture/app/actions.go]: fixed bug

File: ui/src/app/applications/components/utils.tsx [hunk 2]
```diff
@@ -122,7 +122,7 @@ export const ComparisonStatusIcon = ({status}: { status: appModels.SyncStatusCod
 export function syncStatusMessage(app: appModels.Application) {
     let message = '';
     let rev = app.spec.source.targetRevision || 'HEAD';
-    if (app.status.sync.revision.length >= 7 && !app.status.sync.revision.startsWith(app.spec.source.targetRevision)) {
```
Comment (alexec) [inline ui/src/app/applications/components/utils.tsx]: fix for apps that are invalid

File: pkg/apis/application/v1alpha1/types.go [hunk 3]
```diff
@@ -558,8 +558,8 @@ type ComparedTo struct {
 // SyncStatus is a comparison result of application spec and deployed application.
 type SyncStatus struct {
 	Status     SyncStatusCode `json:"status" protobuf:"bytes,1,opt,name=status,casttype=SyncStatusCode"`
-	ComparedTo ComparedTo     `json:"comparedTo" protobuf:"bytes,2,opt,name=comparedTo"`
-	Revision   string         `json:"revision" protobuf:"bytes,3,opt,name=revision"`
+	ComparedTo ComparedTo     `json:"comparedTo,omitempty" protobuf:"bytes,2,opt,name=comparedTo"`
```
Comment (alexec) [inline pkg/apis/application/v1alpha1/types.go]: This is the main change. May break API compatibility but is a bug. @jessesuen

File: test/e2e/app_management_test.go [hunk 4]
```diff
@@ -54,6 +54,16 @@ func TestAppCreation(t *testing.T) {
 		})
 }
 
+func TestInvalidAppProject(t *testing.T) {
```
Comment (alexec) [inline test/e2e/app_management_test.go]: If you create an app via the CLI, it is harder to create a borked app.

File: test/e2e/delarative_test.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		Expect(SyncStatusIs(SyncStatusCodeOutOfSync)).
+		When().
+		Sync().
+		Then().
+		Expect(OperationPhaseIs(OperationSucceeded)).
+		Expect(HealthIs(HealthStatusHealthy)).
+		Expect(SyncStatusIs(SyncStatusCodeSynced))
+}
+
+func TestDeclarativeInvalidPath(t *testing.T) {
```
Comment (alexec) [inline test/e2e/delarative_test.go]: It is easy to create a borked app via `kubctl apply` .

File: test/e2e/delarative_test.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	Given(t).
+		Path("garbage").
+		When().
+		Declarative("declarative-apps/app.yaml").
+		Then().
+		Expect(Success("")).
+		Expect(HealthIs(HealthStatusHealthy)).
+		Expect(SyncStatusIs(SyncStatusCodeUnknown)).
+		When().
+		// TODO - should cascade work here or not?
```
Comment (alexec) [inline test/e2e/delarative_test.go]: I feel like we should be able to delete these apps using cascade deletion. Thoughts @jessesuen

**Review by alexmt — APPROVED** [review]

**Pull request merged**